pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -53,7 +53,13 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	return nil
+	// Watch for changes to secrets being used by hosts
+	err = c.Watch(&source.Kind{Type: &v1.Secret{}},
+		&handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &metalkubev1alpha1.BareMetalHost{},
+		})
+	return err
```

Comment:
If I'm reading this right, it looks like we watch for updates to Secrets owned by a BareMetalHost.  That ownership is set by the BareMetalHost reconciler when it matches them together.

I'm imagining the case where a BareMetalHost is created and refers to a Secret that does not yet exist.  Later, that Secret gets created.  What will ensure that the BMH reconciler runs again to pick up this secret?

or am I misinterpreting the second argument to Watch() ?

Reply:
If the host has no secret or the secret can't be loaded Reconcile() exits with an error and that triggers reconciliation again. That will just keep happening until the secret is there. If we have a secret, and its contents are invalid, then the host is marked with an error and the reconciliation loop stops until the secret reference is changed or the contents of the secret are changed.